modules/output.nix

```diff
@@ -28,6 +28,26 @@ let
 in
 {
   options = {
+    env = mkOption {
+      type =
+        with types;
+        lazyAttrsOf (oneOf [
+          str
+          path
+          int
+          float
+        ]);
```

Comment:
Just a note:

The [nixos impl](https://github.com/NixOS/nixpkgs/blob/3acdee779318c707cfb5735db379b735f314b094/nixos/modules/config/shells-environment.nix#L42-L44) has an `apply` function that coerces path values to `"${value}"` (ensuring they get copied to the nix store) and also does some stuff with list values (although we don't currently support list types).

```nix
apply =
  let
    toStr = v: if lib.isPath v then "${v}" else toString v;
  in
  lib.mapAttrs (n: v: if lib.isList v then lib.concatMapStringsSep ":" toStr v else toStr v);
```